ql/src/test/org/apache/hadoop/hive/ql/metadata/TestHiveMetaStoreChecker.java

```diff
@@ -252,37 +241,165 @@ public void testInvalidPartitionKeyName()
+    addFolderToPath(fs, newPart.toString(), "delta_0000003_0000003_0000");
+    addFolderToPath(fs, newPart.toString(), "base_0000003_v0000200");
+    CheckResult result = checker.checkMetastore(catName, dbName, tableName, null, null);
+    assertEquals(Collections.<CheckResult.PartitionResult> emptySet(), result.getPartitionsNotOnFs());
+    assertEquals(1, result.getPartitionsNotInMs().size());
+    // Found the highest writeId
+    assertEquals(3, result.getPartitionsNotInMs().iterator().next().getMaxWriteId());
+    assertEquals(200, result.getPartitionsNotInMs().iterator().next().getMaxTxnId());
+  }
+
+  @Test
+  public void testAddPartitionMMBase() throws Exception {
+    Table table = createTestTable(true);
+    List<Partition> partitions = hive.getPartitions(table);
+    assertEquals(2, partitions.size());
+    // add a partition dir on fs
+    fs = partitions.get(0).getDataLocation().getFileSystem(hive.getConf());
+    Path newPart = addFolderToPath(fs, table.getDataLocation().toString(),
+        partDateName + "=2017-01-01/" + partCityName + "=paloalto");
+
+    // Add a few deltas
+    addFolderToPath(fs, newPart.toString(), "delta_0000001_0000001_0000");
+    addFolderToPath(fs, newPart.toString(), "delta_0000002_0000002_0000");
+    addFolderToPath(fs, newPart.toString(), "delta_0000003_0000003_0000");
+    addFolderToPath(fs, newPart.toString(), "base_0000004");
+    CheckResult result = checker.checkMetastore(catName, dbName, tableName, null, null);
+    assertEquals(Collections.<CheckResult.PartitionResult> emptySet(), result.getPartitionsNotOnFs());
+    assertEquals(1, result.getPartitionsNotInMs().size());
+    // Found the highest writeId
+    assertEquals(4, result.getPartitionsNotInMs().iterator().next().getMaxWriteId());
+    assertEquals(0, result.getPartitionsNotInMs().iterator().next().getMaxTxnId());
+  }
+
+  @Test
+  public void testNoNPartitionedTable() throws Exception {
+    Table table = createNonPartitionedTable();
+    // add a partition dir on fs
+    fs = table.getDataLocation().getFileSystem(hive.getConf());
+
+    Path tablePath = table.getDataLocation();
+
+    // Add a few deltas
+    addFolderToPath(fs, tablePath.toString(), "delta_0000001_0000001_0000");
+    addFolderToPath(fs, tablePath.toString(), "delta_0000002_0000002_0000");
+    addFolderToPath(fs, tablePath.toString(), "delta_0000003_0000003_0000");
+    addFolderToPath(fs, tablePath.toString(), "base_0000003_v0000200");
+    CheckResult result = checker.checkMetastore(catName, dbName, tableName, null, null);
+    assertEquals(Collections.<CheckResult.PartitionResult> emptySet(), result.getPartitionsNotOnFs());
+    assertEquals(Collections.<CheckResult.PartitionResult> emptySet(), result.getPartitionsNotInMs());
+    // Found the highest writeId
+    assertEquals(3, result.getMaxWriteId());
+    assertEquals(200, result.getMaxTxnId());
+  }
+
+  private Path addFolderToPath(FileSystem fs, String rootPath, String folder) throws IOException{
```

Comment:
could you please move helper methods at the bottom of the Test class after all tests

Reply:
fixed